client/src/components/AddCard/AddCardForm/AddCardForm.tsx

```diff
@@ -0,0 +1,122 @@
+import { CircularProgress } from '@material-ui/core';
+import { addCard } from '../../../helpers/APICalls/addCard';
+import { getAllPaymentMethods } from '../../../helpers/APICalls/getAllPaymentMethods';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+import { useHistory } from 'react-router-dom';
+import CreditCard from './CreditCard/CreditCard';
+
+interface Props {
+  handleSubmit: (
+    {
+      paymentMethod,
+    }: {
+      paymentMethod: string;
+    },
+    {
+      setStatus,
+      setSubmitting,
+    }: FormikHelpers<{
+      paymentMethod: string;
+    }>,
+  ) => void;
+}
+
+export default function AddCard({ handleSubmit }: Props): JSX.Element {
+  const [paymentMethods, setPaymentMethods] = useState<any>([]);
+  const { loggedInUser, loggedInUserProfile } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
+  const history = useHistory();
+
+  useEffect(() => {
+    if (loggedInUser) {
+      getAllPaymentMethods({ userId: loggedInUser.id }).then((data) => {
+        if (data.error) {
+          updateSnackBarMessage(data.error.message);
+        } else if (data.allPaymentMethods) {
+          setPaymentMethods(data.allPaymentMethods);
+        } else {
+          updateSnackBarMessage('An unexpected error occurred. Please try again');
+        }
+      });
+    }
+  }, [loggedInUser, updateSnackBarMessage]);
+
+  const handleAddCard = () => {
+    if (loggedInUser) {
+      addCard({ userId: loggedInUser.id }).then((data) => {
+        if (data.error) {
+          updateSnackBarMessage(data.error.message);
+        } else if (data.customer) {
+          history.push({
+            pathname: '/add-card-info',
+            state: {
+              clientSecret: data.intent.client_secret,
+            },
+          });
+        } else {
+          updateSnackBarMessage('An unexpected error occurred. Please try again');
+        }
+      });
+    }
+  };
+  const classes = useStyles();
+
+  if (process.env.REACT_APP_PRICE_ID) {
+    return (
+      <Formik
+        initialValues={{
+          paymentMethod: loggedInUserProfile ? loggedInUserProfile.defaultPaymentMethod : '',
+        }}
+        validationSchema={Yup.object().shape({
+          paymentMethod: Yup.string().required('PaymentMethod is required'),
+        })}
+        onSubmit={handleSubmit}
+      >
+        {({ handleSubmit, setValues, values, touched, errors, isSubmitting }) => (
+          <form onSubmit={handleSubmit} className={classes.form} noValidate>
+            <Box width="100%" display="flex" flexDirection="column" alignItems="center">
+              <Box maxWidth="85%">
+                <Typography component="h4" variant="h6" className={classes.smallTitle}>
+                  saved payment profiles:
+                </Typography>
+                <Grid container>
+                  {paymentMethods.map((card: any, index: number) => (
+                    <CreditCard
+                      key={index}
```

Comment:
Same comment as previous PR about usage of indexes as keys

Reply:
OK. Done.